src/components/MegaButton/MegaButton.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  <div>
+    <input
+      :id="id"
+      type="radio"
+      class="hidden peer"
+      :name="name"
+      :value="value.toString()"
+      :checked="checked"
+      :disabled="disabled"
+      :readonly="readonly"
+      :required="required"
+      @input="onInput"
+      @click="onClick"
+    >
+    <label
+      :for="id"
+      :class="[
+        'h-full',
+        'flex items-center rounded-lg m-5 inline-block border-4 border-transparent relative shadow-input cursor-pointer',
+        'peer-focus:ring-primary-100 peer-focus:ring-4',
+        'peer-hover:shadow',
+        'peer-checked:border-2 peer-checked:border-primary-500',
+        'peer-disabled:cursor-not-allowed peer-disabled:shadow-none peer-disabled:text-gray-100',
+        'peer-disabled:border-2 peer-checked:border-disabled-gray',
+        { 'strikethru-line': disabled && !disabledBanner }
+      ]"
+    >
+      <div
+        v-if="hasDisabledBanner"
+        class="absolute mt-0  text-center bg-turquoise-100 uppercase text-gray-700 text-sm tracking-wide font-bold p-1 inset-x-0 top-0"
+      >
+        {{ disabledBanner }}
+      </div>
+
+      <div>
+        <div :class="['mx-8', { 'mt-8 mb-9': !hasDisabledBanner }, {'mt-12 mb-6': hasDisabledBanner }]">
+          <div class="font-semibold text-center text-3xl">
+            <slot name="label">
+              {{ label }}
+            </slot>
+          </div>
+        </div>
+      </div>
+    </label>
+  </div>
+</template>
+
+<script>
+import { getCurrentInstance } from 'vue';
+
+export default {
+  name: 'RadioButton',
+  props: {
+    id: {
+      type: String,
+      required: true
+    },
+    modelValue: {
+      type: [String, Boolean],
+      default: null
+    },
+    name: {
+      type: String,
+      default: ''
+    },
+    value: {
+      type: [String, Boolean],
+      default: ''
+    },
+    label: {
+      type: String,
+      default: ''
+    },
+    disabledBanner: {
+      type: String,
+      default: null
+    },
+    required: {
+      type: Boolean,
+      default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    readonly: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: ['update:modelValue', 'input', 'click'],
+  data () {
+    return {
+      parent: null
+    };
+  },
+  computed: {
+    checked () {
+      return this.modelValue === this.value;
+    },
+    hasDisabledBanner () {
+      return this.disabled && this.disabledBanner;
+    }
+  },
+  created () {
+    this.parent = getCurrentInstance().parent;
```

Comment:
what are we using the `parent` for?

Reply:
I think you can do `this.$parent` instead to get easier access to the parent but I think this might be a pattern we should avoid.  The child shouldn't have knowledge of or coupling to it's parent and should be making calls directly on the parent but rather should just emit when events happen and let the parent decide how it wants to respond.

Reply:
agreed with the above, you can do `$this.parent` but it's something we should avoid as much as possible. I believe it's used in our current `RadioButton` to detect if a `sameLine` prop has been passed through to the `RadioGroup` and adjust the radio button's styles accordingly, but from the mocks, this doesn't seem like a situation we need to account for here.